### PR TITLE
#97 #105 Study group form

### PR DESCRIPTION
**Pull request by ONourry**
1- Created the study group form
2- Added the necessary routing for the new form
3- Added a navbar button to allow the user to create study groups
3- Modified the registration form to prevent page refresh (makes the website faster)

The validation for the study group only requires the user to have every field filled out. No specific format is required for the group name, teacher name, course tag or description.

### CODE REVIEW

**Comment by ONourry** [comment]
Seems like travis is mad about something. Let me look at it first.

**Comment by ONourry** [comment]
Alright the build is passing now. I'm not sure why that one line redirecting the url changed the return value but it did so I removed it.

**Comment by simon-bourque** [comment]
I get this error message whenever I try to add a study group
image

**Comment by ONourry** [comment]
I'm not getting any error. What steps are you taking exactly?

**Comment by simon-bourque** [comment]
1. Log in
2. Click on the create study group
3. fill in form with the following
group name : test
course: coen346
teacher: bob bob
description: asdadadad
4. click create study group
5. click ok on alert
6. observe error message appear on console

**Comment by simon-bourque** [comment]
its weird because I don't see how addUser is being called?

**Comment by simon-bourque** [comment]
perhaps add user is being called instead of addUserStudyGroups

File: client/modules/User/UserActions.js [hunk 1]
```diff
@@ -127,3 +128,26 @@ export function logoutUserRequest() {
...
+  browserHistory.replace('/profile');
+  return {
+    type: CREATE_GROUP,
+    studyGroup,
+  };
+}
+
+export function createStudyGroupRequest(studyGroup) {
+  return (dispatch) => {
+    return callApi('users', 'post', {
```
Comment (simon-bourque) [inline client/modules/User/UserActions.js]: should be 'studyGroups' and not 'users' as discussed on slack

**Comment by simon-bourque** [comment]
alright ima test it

**Comment by simon-bourque** [comment]
Approving for sprint 3, we will make a separate issue for what was discussed

**Pull request merged**